src/main/java/org/broadinstitute/hellbender/engine/FeatureManager.java

```diff
@@ -208,6 +211,20 @@ public boolean isEmpty() {
         return featureSources.isEmpty();
     }
 
+
+    /**
+     *
+     */
+    public List<VCFHeader> getAllVariantHeaders() {
+        List<VCFHeader> headers = new ArrayList<>();
+        for (Map.Entry<FeatureInput<? extends Feature>, FeatureDataSource<? extends Feature>> entry : featureSources.entrySet()) {
+            if (entry.getKey().getFeatureType() == VariantContext.class) {
+                headers.add((VCFHeader)entry.getValue().getHeader());
```

Comment:
Add a protective check here, just in case there are codecs that produce `VariantContext`s but do not have a header in vcf format:

        final Object header = entry.getValue().getHeader();
        if ( header instanceof VCFHeader ) {
            headers.add((VCFHeader)header);
        }